src/librustc_lint/builtin.rs

```diff
@@ -41,6 +41,7 @@ use lint::{Level, Context, LintPass, LintArray, Lint};
 
 use std::collections::BitSet;
 use std::collections::hash_map::Entry::{Occupied, Vacant};
+#[allow(deprecated)]
 use std::num::SignedInt;
```

Comment:
Actually there's a bunch of `#[allow(deprecated)]` imports of these traits, but I'm not sure they all correspond to a generic bound, so it seems like they should be able to be removed?